lib/flowViewport/API/perViewportSceneIndicesData/fvpViewportInformationAndSceneIndicesPerViewportData.cpp

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright 2023 Autodesk, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//Local headers
+#include "fvpViewportInformationAndSceneIndicesPerViewportData.h"
+#include "flowViewport/API/interfacesImp/fvpInformationInterfaceImp.h"
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace FVP_NS_DEF {
+
+ViewportInformationAndSceneIndicesPerViewportData::ViewportInformationAndSceneIndicesPerViewportData(const CreationParameters& creationParams)
+    : _viewportInformation(creationParams._viewportInformation), _renderIndexProxy(creationParams._renderIndexProxy)
+{
+}
+
+ViewportInformationAndSceneIndicesPerViewportData::~ViewportInformationAndSceneIndicesPerViewportData()
+{
+    //The viewport info is own by this class
+    InformationInterfaceImp::Get().SceneIndexRemoved(_viewportInformation);
+    delete &_viewportInformation;
```

Comment:
It is created by MayaHydraSceneProducer::MayaHydraSceneProducer in :
FVP_NS_DEF::InformationInterface::ViewportInformation* hydraViewportInformation = 
    new FVP_NS_DEF::InformationInterface::ViewportInformation(_sceneIndex, initData.cameraName, initData.viewportWidth, initData.viewportHeight, initData.rendererName);
 
So it's on the heap and nothing else deletes it.

Reply:
Again, an explicit delete violates
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-newdelete

Reply:
I removed this delete by storing copies instead of pointers as the class is not heavy and the size of the std set will be the number of hydra viewports in maya